README.rst

```diff
@@ -1,95 +1,226 @@
 This pattern is typical in multiple inheritance.
 
 We hope that Python linters will be able to check for the super call.
 
 Augmenting is declared using two decorators:
 
-* ``augments`` indicates that this method must call super within its definition and thus augments the behavior of the base class method, and
-* ``must_agugment`` indicates that child classes that define this method must decorate their method overriddes with ``augments``.
+* ``augments`` indicates that this method must call super within its definition
+  and thus augments the behavior of the base class method, and
+* ``must_augment`` indicates that child classes that define this method must
+  decorate their method overriddes with ``augments``.
 
-For example:
+From ``samples/augments.py`` :
 
 .. code-block:: python
 
-    class HasMustAugmentMethod(AbstractBaseClass):
+    from ipromise import AbstractBaseClass, must_augment, augments
+    import abc
+
+    class MyClass(AbstractBaseClass):
         @must_augment
         def f(self):
             # must_augment prevents this behavior from being lost.
+            print("MyClass().f()")
             self.times_f_called += 1
             return 0
 
-
-    class AugmentsMethod(HasMustAugmentMethod):
-        @augments(HasMustAugmentMethod)
+    class MyClassAgumentedOnce(MyClass):
+        @augments(MyClass)
         def f(self, extra=0, **kwargs):
+            print("MyClassAgumentedOnce().f()")
             return super().f(**kwargs) + extra
 
-
-    class AugmentsMethodFurther(AugmentsMethod):
-        @augments(HasMustAugmentMethod)
+    class MyClassAgumentedOnceAgain(MyClassAgumentedOnce):
+        @augments(MyClass)
         def f(self, **kwargs):
-            print("f has been called")
+            print("MyClassAgumentedOnceAgain().f()")
             return super().f(**kwargs)
+
+    MyClassAgumentedOnce().f()
+    MyClassAgumentedOnceAgain().f()
+
+***********
+Development
+***********
+
+Pull Requests can be submitted on github.
+
+poetry
+======
+
+The poetry development environment can be started with the typical
+poetry commands:
+
+.. code-block:: shell
+
+    poetry install
+    poetry shell
+
+build failure
+-------------
+
+Command ``poetry install`` may report a failure to build all pip packages but
+the command will succeed (return ``0``). In that case, users on apt-managed
+Linux may need to install extra apt packages then run the command again:
+
+.. code-block:: shell
+
+    apt install libffi-dev libxml2 libxslt-dev
+
+*MMV*
+
+Tools
+=====
+
+Tool commands should be run at the project top-level directory.
+
+mypy
+----
+
+.. code-block:: shell
+
+    mypy ipromise test
+
+flake8
+------
+
+.. code-block:: shell
+
+    pflake8 ipromise test
+
+pytest
+------
+
+.. code-block:: shell
+
+    pytest -c pyproject.toml test
+
+rst-lint
+--------
+
+Only necessary for ``README.rst`` changes.
+
+.. code-block:: shell
+
+    pip install restructuredtext_lint Pygments
```

Comment:
This install should be in pyproject.  If it isn't working for you, there's something wrong with your poetry setup.

Reply:
Fixed.